pkg/certs/cert-inspection/certgraphanalysis/metadata_options.go

```diff
@@ -43,4 +52,41 @@ var (
 			}
 		},
 	}
+	RewriteNodeIPs = &metadataOptions{
+		rewriteCertKeyPair: func(certKeyPair *certgraphapi.CertKeyPair, o rewriteParams) {
+			rewriteIPInEtcdServingMetricsCertificateName(certKeyPair, o.nodes)
+			rewriteIPInEtcdServingCertificate(certKeyPair, o.nodes)
+			rewriteIPInEtcdPeerCertificate(certKeyPair, o.nodes)
+		},
+	}
 )
+
+func rewriteIPInEtcdServingCertificate(certKeyPair *certgraphapi.CertKeyPair, nodes map[string]int) {
```

Comment:
Let's talk at 10.  I think you've discovered two different mutators that need to happen at two different times.

1. secret itself requiring mutation so that the list of "these secret namespace,name tuples are present" is correct when stored.
2. the certkeypair and cabundle metadata so that names are consistent.

1 happens before filtering in the first loop
2 happens here.

Reply:
In this case rewriting secret/configmap is sufficient